src/dom.js

```diff
@@ -13,218 +13,27 @@
  * License for the specific language governing permissions and limitations under
  * the License.
  */
-
-import parse5 from 'parse5';
-import nwmatcher from 'nwmatcher';
-
-// htmlparser2 has a relatively DOM-like tree format, which we'll massage into a DOM elsewhere
-const treeAdapter = parse5.treeAdapters.htmlparser2;
-
-const PARSE5_OPTS = {
-  treeAdapter
-};
+import {JSDOM} from 'jsdom';
 
 /**
  * Parse HTML into a mutable, serializable DOM Document.
  * The DOM implementation is an htmlparser2 DOM enhanced with basic DOM mutation methods.
  * @param {String} html   HTML to parse into a Document instance
  */
 export function createDocument (html) {
-  const document = parse5.parse(html, PARSE5_OPTS);
-
-  defineProperties(document, DocumentExtensions);
-  // Find the first <html> element within the document
-
-  // Extend Element.prototype with DOM manipulation methods.
-  //   Note: document.$$scratchElement is also used by createTextNode()
-  const scratch = document.$$scratchElement = document.createElement('div');
-  const elementProto = Object.getPrototypeOf(scratch);
-  defineProperties(elementProto, ElementExtensions);
-  elementProto.ownerDocument = document;
-
-  // nwmatcher is a selector engine that happens to work with Parse5's htmlparser2 DOM (they form the base of jsdom).
-  // It is exposed to the document so that it can be used within Element.prototype methods.
-  document.$match = nwmatcher({ document });
-  document.$match.configure({
-    CACHING: false,
-    USE_QSAPI: false,
-    USE_HTML5: false
+  const { window } = new JSDOM(html, {
+    contentType: "text/html",
   });
 
+  const document = window.document;
+
   return document;
 }
-
 /**
  * Serialize a Document to an HTML String
  * @param {Document} document   A Document, such as one created via `createDocument()`
  */
 export function serializeDocument (document) {
-  return parse5.serialize(document, PARSE5_OPTS);
+  return document.querySelector('html').innerHTML;
```

Comment:
think this skips serialization of the doctype.
the JSDOM instance from line 24 has a `.serialize()` method that does this a little quicker and includes that info.

```js
  const jsdom = new JSDOM(html, {
    contentType: "text/html",
  });
  const { window } = jsdom;
  window.$jsdom = jsdom;
}
export function serializeDocument(document) {
  document.defaultView.$jsdom.serialize(document);
}
```